lib/utils/report.mjs

```diff
@@ -51,7 +54,7 @@ export default function report(problem) {
 	// In quiet mode, mere warnings are ignored
 	if (quiet && ruleSeverity === SEVERITY_WARNING) return;
 
-	if (isFn(fix) && metadata && !metadata.fixable) {
+	if ((isFn(fix) || isFixObject(fix)) && metadata && !metadata.fixable) {
 		throw new Error(
 			`The "${ruleName}" rule requires "meta.fixable" to be truthy if the "fix" callback is being passed`,
```

Comment:
minor

This line will have to be updated to cover both `fix` and `fix.apply`.

Reply:
I think `if the "fix" callback is being passed` also covers `fix.apply` :)

We mainly want to warn that `meta.fixable` needs to be set when we detect a correct "fixer". We don't have to super technical in how we describe that fixer.